pynitrokey/fido2/client.py

```diff
@@ -34,39 +34,39 @@
 
 
 class NKFido2Client:
-    def __init__(
-        self,
-    ):
+    def __init__(self):
+
         self.origin = "https://example.org"
         self.host = "example.org"
         self.user_id = b"they"
         self.exchange = self.exchange_hid
         self.do_reboot = True
 
-    def use_u2f(
-        self,
-    ):
+    def __getattr__(self, key: str) -> Any:
+        """Reflecting self.client into self"""
+
+        if hasattr(self.client, key):
+            return getattr(self.client, key)
+        raise AttributeError(key)
```

Comment:
Also, does this affect the type checks or can mypy understand this reflection?

Reply:
This is the typical python-`__getattr__` pattern to apply, but keep in mind that `__getattr__` is only invoked if **no other** member/attribute within this object's `__ dict__` matches the `key`, thus it is best practice for the method to behave as if you would do a regular object's attribute lookup. (otherwise you can easily build funny endless-lookup-loops), On a side note: there is also `__getattribute__` , which will behave as you expected and intercepts every attribute-lookup.

clear trade-off here: 
* either write considerably less code, with the typical (unsafe) implicit-magic such mechanisms come with
* or be very explicit and thus far more precise / less error prone

we don't have to decide this now, this might come handy once we slim down our `fido2` class here as this then more-or-less works like a mixin. These lines took longer than this code, so I don't care let's vote yay or nay in this comment!

Reply:
left it out for now